Add column search, regex search

egisz opened the pull request:
Hi,

I have added:

- individual column search (fixes Issue #9), 
- regex search which is useful with column search, check [example](http://live.datatables.net/kopafape/1/edit).
- support for sequelize v4, v5.
- tests for column search, mixed global and column search, regex search.

@alwint3r, could you publish to npmjs, so that we could include to our projectsr (or grant me permissions)?
I would like to avoid publishing my own npm as duplicate.

alwint3r commented:
First of all, this is great! This should solve the compatibility issue with newer versions of sequelize.
However, i can not test this using PostgreSQL database due to PostgreSQL driver installation failure. Probably it's because of my messed-up OS installation. Other than that, all tests are passed.

Before merging this PR and publish the new version to npm, maybe we should state that node 4 is no longer supported in the README.md since there are async functions all over the place. 
Since we're dropping support for older version of node, should we bump the major version instead of minor? Although there's no major change in the API. I'd like to hear your thought on it.

Thanks a lot!

egisz commented:
Hi,
It's good that you mentioned tests, I double checked and fixed one bug. I missed that datatables  search.regex parameter is passed as string with value `true` or `false`. I expected boolean :)
Also fixed test with case insensitive search, it should fail on mysql  
I have tested on PostgreSQL, all tests now passed.

![image](https://user-images.githubusercontent.com/1941760/82748564-9c91f100-9dab-11ea-9ff6-bb4f1afb0df8.png)

> Before merging this PR and publish the new version to npm, maybe we should state that node 4 is no longer supported in the README.md since there are async functions all over the place.
Good point, updated README. in last commit.

> Since we're dropping support for older version of node, should we bump the major version instead of minor? Although there's no major change in the API. I'd like to hear your thought on it.
You are right, I was very confused regarding versions. In npmjs there is another published package https://www.npmjs.com/package/sequelize-datatable and it is already claiming version 2.0.0.
So now it would be wise to set it higher version as this is more fresh and it has more features.
This is why I thought of version 2.1.0. But we could easily name it 3.0.0 as well. I just want to avoid mess in npmjs, so that developers could easily find needed module.

alwint3r commented on CHANGELOG.md:
```diff
@@ -1,40 +1,35 @@
 ===============
 
-All notable changes to this project will be documented in this file. This project adheres to [Semantic Versioning](http://semver.org/).
+### v2.1.0-0 - 2020-05-18
```
Maybe we should change this to 3.0.0 instead of 2.1.0

egisz replied:
Updated in the last commit.
